packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/FlipbookViewer/FlipbookViewerContainer.js

```diff
@@ -55,6 +55,18 @@ function FlipbookViewerContainer({
    * We're assuming all frames in one subject have the same dimensions. */
   const defaultFrameUrl = subject ? Object.values(subject.locations[defaultFrame])[0] : null
   const { img, error, loading } = useSubjectImage(defaultFrameUrl)
+
+  useEffect(function preloadImages() {
+    subject?.locations?.forEach(location => {
+      const [url] = Object.values(location)
+      const { Image } = window
+      const img = new Image()
+      if (url) {
+        img.src = url
+      }
+    })
+  }, [])
```

Comment:
Getting closer, but the preloader needs to run for each new subject. This only runs for the first subject. React should warn about missing effect dependencies here. 
```suggestion
  useEffect(function preloadImages() {
    subject?.locations?.forEach(location => {
      const [url] = Object.values(location)
      if (url) {
        const { Image } = window
        const img = new Image()
        img.src = url
      }
    })
  }, [subject?.locations])
```

Reply:
Because `subject` is a prop of FlipbookViewerContainer, when `subject` changes during loading of a new subject, the component calls this `useEffect()` again. For instance, if I log to the console from inside the preloading `forEach()` loop, and then complete classifications on a project, the message is logged to the console every time a new subject loads.

Is that a different props flow then you were thinking?

Reply:
Setting the effects dependency array to `[]` tells it to ignore prop and state changes, so this hook should only run once, and ignore `subject` changes.

Reply:
Is the subject viewer keyed by subject ID? That would mount this component for each new subject, in which case the hook will run for each new subject (but React will still warn you that the effect hasn't declared its dependencies.)

Reply:
Aha! A new subject viewer mounts for each subject. I wonder why we did that?🤔
https://github.com/zooniverse/front-end-monorepo/blob/db4698b06379d57d94c69d554605aafba61ff87c/packages/lib-classifier/src/components/Classifier/components/SubjectViewer/SubjectViewer.js#L54-L61

Reply:
Workaround for a PH-TESS bug (which we still see logged in Sentry.)
https://github.com/zooniverse/front-end-monorepo/pull/720

Reply:
Interesting... I'll still add your suggested commit though for best practices of including dependencies in the useEffect array.

Reply:
This does mean viewer component state won’t persist across subjects. I don’t know if that’s good or bad. Not an issue for this PR though.